src/game/boe.actions.cpp

```diff
@@ -2976,10 +2977,10 @@ bool town_move_party(location destination,short forced) {
 		forced = true;
 	if(univ.debug_mode && univ.ghost_mode)
 		forced = keep_going = true;
-	
-	ter = univ.town->terrain(destination.x,destination.y);
-	
+
 	if(keep_going) {
+		// ASAN destination is only correct if keep_going=true
+		ter = univ.town->terrain(destination.x,destination.y);
```

Comment:
Though it's correct to move this here, how can it solve an ASAN problem? There's nothing here or in `check_special_terrain` that verifies the location is actually on the map.

Reply:
Yes, it would be better to check that the destination is valid at the beginning of this function.

( I guess when asan detected a problem here, keep_going was false so I made this change but it is not a good change )

Reply:
I've added a bounds check both here and in `outd_move_party`.